frontend/src/metabase/embedding/lib/override-requests-for-embeds.ts

```diff
@@ -0,0 +1,265 @@
+  const matchedPattern = findMatchingPattern(url);
+
+  if (
+    getIgnoreOverridePatterns().some(
+      (endpoint) => endpoint && url.includes(endpoint),
+    )
+  ) {
+    return null;
+  }
+
+  // No transformation needed if pattern doesn't match
+  if (!matchedPattern) {
+    return { method, url, options };
+  }
+
+  // Apply the transformation for this pattern
+  const transformation = EMBED_URL_TRANSFORMATIONS[matchedPattern]({
+    embedType,
+  });
+  if (!transformation) {
+    return { method, url, options };
+  }
+
+  return {
+    ...transformation,
+    options: {
+      ...options,
+      hasBody: transformation.method === "POST",
+    },
+  };
+}
+
+/**
+ * Replaces the standard API base path with the embed API base path.
+ */
+function replaceWithEmbedBase({
+  embedType,
+  url,
+}: {
+  embedType: EmbedType;
+  url: string;
+}): string {
+  const baseUrl = getBaseUrlByEmbedType(embedType);
+
+  if (url.includes(internalBase) && !url.includes(baseUrl)) {
+    return url.replace(internalBase, baseUrl);
+  }
+
+  return url;
+}
+
+const overrideRequests = async ({
+  embedType,
+  method,
+  url,
+  options,
+}: OnBeforeRequestHandlerData & {
+  embedType: EmbedType;
+}) => {
+  const transformation = getRequestTransformation({
+    method,
+    embedType,
+    url,
+    options,
+  });
+
+  if (!transformation) {
+    return { method, url, options };
+  }
+
+  if (!options.headers) {
+    options.headers = {};
+  }
+
+  /**
+   * Set header to indicate that this request is for guest embed.
+   */
+  options.headers["x-metabase-guest-embed"] = "true";
+
+  return {
+    method: transformation.method,
+    url: replaceWithEmbedBase({ embedType, url: transformation.url }),
+    options: transformation.options,
+  };
+};
+
+const setupRemappingUrls = (embedType: EmbedType) => {
+  const baseUrl = getBaseUrlByEmbedType(embedType);
+
+  PLUGIN_API.getRemappedDashboardParameterValueUrl = (
+    _dashboardId: DashboardId | undefined,
+    parameterId: ParameterId,
+  ) =>
+    `${baseUrl}/dashboard/:entityIdentifier/params/${encodeURIComponent(parameterId)}/remapping`;
+
+  PLUGIN_API.getRemappedCardParameterValueUrl = (
+    _cardId: CardId | string | undefined,
+    parameterId: ParameterId,
+  ) =>
+    `${baseUrl}/card/:entityIdentifier/params/${encodeURIComponent(parameterId)}/remapping`;
+};
+
+/**
+ * Registers a request interceptor that transforms standard API requests
+ * into guest embeds API requests.
+ */
+export const overrideRequestsForGuestEmbeds = () => {
+  setupRemappingUrls("guest");
+
+  PLUGIN_EMBEDDING_SDK.onBeforeRequestHandlers.overrideRequestsForGuestEmbeds =
+    (data) =>
+      overrideRequests({
+        ...data,
+        embedType: "guest",
+      });
+};
+
+/**
+ * Registers a request interceptor that transforms standard API requests
+ * into public or static embeds API requests.
+ */
+export const overrideRequestsForPublicOrStaticEmbeds = (
+  embedType: EmbedType,
```

Comment:
This shouldn't be `EmbedType` as it also includes `guest`. Here we probably only accepts only 2 values `public` or `static`